src/rod/builder/primitives.py

```diff
@@ -54,3 +59,41 @@ def _geometry(self) -> rod.Geometry:
         return rod.Geometry(
             cylinder=rod.Cylinder(radius=self.radius, length=self.length)
         )
+
+
+@dataclasses.dataclass
+class MeshBuilder(PrimitiveBuilder):
+    mesh_path: Union[str, pathlib.Path]
+    scale: NDArray
+
+    def __post_init__(self) -> None:
+        """
+        Post-initialization method for the class.
+        Loads the mesh from the specified file path and performs necessary checks.
+
+        Raises:
+            AssertionError: If the scale is not a 3D vector.
+        """
+
+        self.mesh: trimesh.base.Trimesh = trimesh.load(
+            str(self.mesh_path),
+            file_type="stl",
```

Comment:
Can we extract automatically the extension of the file from `mesh_path`? You can use [`pathlib.PurePath.suffix`](https://docs.python.org/3/library/pathlib.html#pathlib.PurePath.suffix). I assume that trimesh already fails if it receives an unsupported file type.